### PR TITLE
Refactor to pep 604

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Changed

- Refactor to pep 604

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor-to-PEP-604 -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by islean — APPROVED** [review]
Looks nice!

**Comment by henrikstranneheim** [comment]
@islean Good catch 👍

**Review by diitaz93 — APPROVED** [review]
💯

File: trailblazer/store/crud/read.py [hunk 1]
```diff
@@ -40,14 +40,14 @@ def get_nr_jobs_with_status_per_category(
...
-    ) -> Optional[list[Analysis]]:
+        before: datetime | None = None,
+        comment: str | None = None,
+        status: str | None = None,
+    ) -> list[Analysis] | None:
         """Return analyses meeting supplied arguments."""
         if not before and not comment and not status:
             return
-        filter_map: dict[Callable, Optional[Union[str, datetime, TrailblazerStatus]]] = {
+        filter_map: dict[Callable, str | datetime | TrailblazerStatus] | None = {
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply trailblazer/store/crud/read.py]: 🦅

**Comment by henrikstranneheim** [comment]
### CLI and API ✅ 

```
[henrik.stranneheim@hasta:~] [S_base] 2m57s $ trailblazer scan
Error updating analysis for: case - maturejay : FileNotFoundError
Error updating analysis for: case - maturejay : FileNotFoundError
Error updating analysis for: case - greatsnipe : FileNotFoundError
Error updating analysis for: case - grandaphid : PermissionError
Error updating analysis for: case - goodbengal : MissingSqueueOutput
Error updating analysis for: case - helpedlab : MissingSqueueOutput
Error updating analysis for: case - newgopher : PermissionError
Error updating analysis for: case - onestag : PermissionError
Error updating analysis for: case - funnyaphid : FileNotFoundError
Error updating analysis for: case - proudfish : PermissionError
Error updating analysis for: case - setcolt : PermissionError
Error updating analysis for: case - handyshrimp : FileNotFoundError
Error updating analysis for: case - divinetomcat : PermissionError
Error updating analysis for: case - HW2V2DRX2 : FileNotFoundError
Error updating analysis for: case - savingstag : FileNotFoundError
Error updating analysis for: case - topewe : PermissionError
Error updating analysis for: case - fullgoat : PermissionError
Error updating analysis for: case - bolddrum : PermissionError
Error updating analysis for: case - abovehusky : PermissionError
Error updating analysis for: case - chiefgannet : PermissionError
Error updating analysis for: case - grandoyster : PermissionError
Status in Tower: coolkodiak - 282044 - WPjrjs23yxQwq
Error logging case - coolkodiak :  JSONDecodeError
Status in Tower: truepigeon - 282656 - 1OGx9sb0sAuwRV
Error logging case - truepigeon :  ValidationError
Status in Tower: fluentgull - 282661 - 26fRfyx4OgMLkn
Error logging case - fluentgull :  ValidationError
Updated status acetuna - 282676: failed
Updated status casualweasel - 282677: completed
Updated status eagerox - 282678: failed
Updated status setamoeba - 282679: failed
Updated status unitedbeagle - 282680: failed
Updated status uphippo - 282681: failed
All analyses updated!
```

Screenshot 2023-11-22 at 08 28 06

**Pull request merged**